raiden/network/channelgraph.py

```diff
@@ -75,16 +169,90 @@ def get_paths_of_length(self, source, num_hops=1):
             if len(path) == num_hops + 1
         ]
 
-    def has_path(self, source, target):
-        """ Return True if there is a path connecting source and target, False
-        otherwise.
+    def get_best_routes(self, our_address, target_address, amount, lock_timeout=None):
+        """ Yield a two-tuple (path, channel) that can be used to mediate the
+        transfer. The result is ordered from the best to worst path.
         """
-        return networkx.has_path(self.graph, source, target)
+        available_paths = self.get_shortest_paths(
+            our_address,
+            target_address,
+        )
+
+        # XXX: consider using multiple channels for a single transfer. Useful
+        # for cases were the `amount` is larger than what is available
+        # individually in any of the channels.
+        #
+        # One possible approach is to _not_ filter these channels based on the
+        # distributable amount, but to sort them based on available balance and
+        # let the task use as many as required to finish the transfer.
+
+        for path in available_paths:
+            partner = path[1]
+            channel = self.partneraddress_channel[partner]
+
+            if not channel.isopen:
+                if log.isEnabledFor(logging.INFO):
+                    log.info(
+                        'channel %s - %s is closed, ignoring',
+                        pex(path[0]),
+                        pex(path[1]),
+                    )
+
+                continue
 
-    def add_path(self, from_, to_):
+            # we can't intermediate the transfer if we don't have enough funds
+            if amount > channel.distributable:
+                if log.isEnabledFor(logging.INFO):
+                    log.info(
+                        'channel %s - %s doesnt have enough funds [%s], ignoring',
+                        pex(path[0]),
+                        pex(path[1]),
+                        amount,
+                    )
+                continue
+
+            if lock_timeout:
+                # Our partner wont accept a lock timeout that:
+                # - is larger than the settle timeout, otherwise the lock's
+                #   secret could be released /after/ the channel is settled.
+                # - is smaller than the reveal timeout, because that is the
+                #   minimum number of blocks required by the partner to learn the
+                #   secret.
+                valid_timeout = channel.reveal_timeout <= lock_timeout <= channel.settle_timeout
```

Comment:
What do you mean?

https://github.com/raiden-network/raiden/blob/master/raiden/channel.py#L506 like this?